src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/get_file_request.cc

```diff
@@ -0,0 +1,309 @@
+  LOG4CXX_DEBUG(logger_, "Intialize non manadatory params with default values");
+
+  if ((*message_)[strings::msg_params].keyExists(strings::file_type)) {
+    file_type_ = static_cast<mobile_apis::FileType::eType>(
+        (*message_)[strings::msg_params][strings::file_type].asInt());
+  }
+
+  if ((*message_)[strings::msg_params].keyExists(strings::length)) {
+    length_ = (*message_)[strings::msg_params][strings::length].asInt();
+  }
+
+  if ((*message_)[strings::msg_params].keyExists(strings::offset)) {
+    offset_ = (*message_)[strings::msg_params][strings::offset].asInt();
+  }
+
+  // Check if file exists on system (may have to use app service id to get the
+  // correct app folder)
+  LOG4CXX_DEBUG(logger_, "Check if file exists on system");
+  std::string file_path;
+  bool forward_to_hmi;
+
+  if (GetFilePath(file_path, forward_to_hmi)) {
+    if (forward_to_hmi) {
+      LOG4CXX_DEBUG(logger_, "Forwarding GetFile request to HMI");
+      SendHMIRequest(hmi_apis::FunctionID::BasicCommunication_GetFileFromHMI,
+                     &(*message_)[strings::msg_params],
+                     true);
+      return;
+    }
+  } else {
+    LOG4CXX_ERROR(logger_, "Could not get file path");
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
+                 "Could not get file path",
+                 &response_params);
+  }
+
+  const std::string full_path = file_path + "/" + file_name_;
+  if (!file_system::FileExists(full_path)) {
+    LOG4CXX_ERROR(logger_, "File " << full_path << " does not exist");
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
+                 "File does not exist",
+                 &response_params);
+    return;
+  }
+
+  // Handle offset
+  LOG4CXX_DEBUG(logger_, "Handle offset parameter");
+  const uint64_t file_size = file_system::FileSize(full_path);
+  if (offset_ > file_size) {
+    LOG4CXX_ERROR(logger_,
+                  "Offset " << offset_ << " greater than file size "
+                            << file_size);
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
+                 "Offset greater than file size",
+                 &response_params);
+    return;
+  }
+
+  // Load data from file as binary data
+  LOG4CXX_DEBUG(logger_, "Load binary data from file");
+  std::vector<uint8_t> bin_data;
+  if (!file_system::ReadBinaryFile(full_path, bin_data, offset_)) {
+    LOG4CXX_ERROR(logger_, "Failed to read from file: " << full_path);
+    SendResponse(false,
+                 mobile_apis::Result::GENERIC_ERROR,
+                 "Unable to read from file",
+                 &response_params);
+    return;
+  }
+
+  // Construct response message
+  if ((*message_)[strings::msg_params].keyExists(strings::offset)) {
+    response_params[strings::offset] = offset_;
+  }
+  if ((*message_)[strings::msg_params].keyExists(strings::length)) {
+    response_params[strings::length] = length_;
```

Comment:
If the `length_` differs from the actual data length, the actual data length should be set here